### DISCUSSION TITLE
Doing something in the component on the receipt of the action

### DISCUSSION THREAD

**Discussion by americanslon** [TimeWarpEngineering/blazor-state#278] [Q&A] [answered]
In my component that is subscribed to a state likes so `private ModalState _state => GetState<ModalState>();` I am trying to catch the subscription event and do something else with the parameters coming from the state change.

What's the most appropriate way of doing so?

**Comment by StevenTCramer** [comment]
The state can only change by handling an Action. If you want to do "something else" then I would either update the handler of the Action or raise a INotification and register a notification handler.  Does that make sense?

**Comment by americanslon** [comment, marked as answer]
It does. But I think I am still missing a piece. Consider this though.

I have a DisplaySizeState (using blazor state) that keeps track of display size breakpoints.

Then I have a relatively complex Component that has a bunch of local variables that are meant to be set according to their own logic based on the display size breakpoints. These variables have no context outside this component and setting them all up in individual actions would just be clutter and bad separation imho.

How what I go about dealing with this? Are you saying I should be raising a INotification in the handler and then registering a notification handler in the Component? Isn't that kind of double pipelining (for the lack of better term) if you will?

**Reply by StevenTCramer** [reply]
I agree having notification handler in a component does NOT sound right.

Looking at the component [lifecycle](https://blazor-university.com/components/component-lifecycles/)

![image](https://user-images.githubusercontent.com/357219/155059050-50672da3-05bb-4d0a-9001-091d7a188573.png)


When a State has changed via BlazorState it calls the StateHasChanged method for you automatically.  You can override `ShouldRender` and adjust your local variables there.

Please let me know if that works for you.

**Reply by americanslon** [reply]
Oh yeah, this is significantly more elegant.

Thanks for teaching me something new about blazor.

**Reply by StevenTCramer** [reply]
Thank my good friend Pete who wrote Blazor University for the above diagram :)
